unstable/lens/string.rkt

```diff
@@ -0,0 +1,43 @@
+#lang racket/base
+
+(provide string-ref-lens
+         string-pluck-lens
+         )
+
+(require fancy-app
+         lens/base/main
+         "join.rkt"
+         )
+(module+ test
+  (require rackunit))
+
+(define (string-ref-lens i)
+  (make-lens
+   (string-ref _ i)
+   (string-set _ i _)))
```

Comment:
I think this can just be done by building it atop list-ref lens using the `string->list` and `string` functions, which would be simpler than writing an explicit `string-set` procedure.